src/lib/dialog/README.md

```diff
@@ -0,0 +1,88 @@
+# MdDialog
+
+MdDialog is a service, which opens dialogs components in the view. 
+
+### Methods
+
+| Name |  Description |
+| --- | --- |
+| `open(component: ComponentType<T>, config: MdDialogConfig): MdDialogRef<T>` | Creates and opens a dialog matching material spec. |
+
+### Config
+
+| Key |  Description |
+| --- | --- |
+| `viewContainerRef: ViewContainerRef` | The view container ref to attach the dialog to. |
+| `role: DialogRole = 'dialog'` | The ARIA role of the dialog element. Possible values are `dialog` and `alertdialog`. |
+
+## MdDialogRef
+
+A reference to the dialog created by the MdDialog `open` method.
+
+### Methods
+
+| Name |  Description |
+| --- | --- |
+| `close(dialogResult?: any)` | Closes the dialog, pushing a value to the afterClosed observable. |
+| `afterClosed(): Observable<any>` | Returns an observable which will emit the dialog result, passed to the `close` method above. |
+
+## Example
+The service can be injected in a component.
+
+```ts
+@Component({
+  selector: 'my-component',
+  template: `
+  <button type="button" (click)="openDialog()">Open dialog</button>
+  `
+})
+export class MyComponent {
+
+  dialogRef: MdDialogRef<MyDialog>;
+
+  constructor(
+    public dialog: MdDialog,
+    public viewContainerRef: ViewContainerRef) { }
+
+  openDialog() {
+    let config = new MdDialogConfig();
+    config.viewContainerRef = this.viewContainerRef;
+
+    this.dialogRef = this.dialog.open(MyDialog, config);
+
+    this.dialogRef.afterClosed().subscribe(result => {
+      console.log('result: ' + result);
+      this.dialogRef = null;
+    });
+  }
+}
+
+@Component({
+  selector: 'my-dialog',
```

Comment:
I like to use example components names that are _super obviously_ examples (which is why there are so many `PizzaComponent`s in the source). This avoid the problem of people seeing this selector and thinking it's a magic identifier that you must use. 

Can you change `MyDialog` and `MyComponent` to be something else?